### PR TITLE
drawio: unbreak on darwin

### PR DESCRIPTION
**Pull request by wegank**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/drawio/default.nix [hunk 1]
```diff
@@ -65,6 +71,11 @@ stdenv.mkDerivation rec {
   installPhase = ''
     runHook preInstall
 
+  '' + lib.optionalString stdenv.isDarwin ''
+    mkdir -p $out/{Applications,bin}
+    mv dist/mac*/draw.io.app $out/Applications
+    makeWrapper $out/Applications/draw.io.app/Contents/MacOS/draw.io $out/bin/drawio
+  '' + lib.optionalString stdenv.isLinux ''
```
Comment (alyssais) [inline pkgs/applications/graphics/drawio/default.nix:78]:
```suggestion
  '' + lib.optionalString (!stdenv.isDarwin) ''
```

as we'd do the same thing on BSD etc if our Electron packages supported it.

File: pkgs/applications/graphics/drawio/default.nix [hunk 2]
```diff
@@ -54,9 +54,15 @@ stdenv.mkDerivation rec {
   buildPhase = ''
     runHook preBuild
 
+  '' + lib.optionalString stdenv.isDarwin ''
+    cp -R ${electron}/Applications/Electron.app Electron.app
+    chmod -R u+w Electron.app
```
Comment (alyssais) [inline pkgs/applications/graphics/drawio/default.nix:59]: Why is this required?

Comment (wegank) [reply pkgs/applications/graphics/drawio/default.nix:59]: `electron-builder` appears to build `dist/mac*/draw.io.app` directly on top of `Electron.app` by overwriting the files in the bundle. Not doing this results in an error when manipulating `Electron.app/Contents/Frameworks/Electron Helper.app/Contents/Info.plist`.

File: pkgs/applications/graphics/drawio/default.nix [hunk 3]
```diff
@@ -65,6 +71,11 @@ stdenv.mkDerivation rec {
   installPhase = ''
     runHook preInstall
 
+  '' + lib.optionalString stdenv.isDarwin ''
+    mkdir -p $out/{Applications,bin}
+    mv dist/mac*/draw.io.app $out/Applications
+    makeWrapper $out/Applications/draw.io.app/Contents/MacOS/draw.io $out/bin/drawio
```
Comment (alyssais) [inline pkgs/applications/graphics/drawio/default.nix:77]: `makeWrapper` with no arguments seems a bit strange — why not just a symlink?

Comment (wegank) [reply pkgs/applications/graphics/drawio/default.nix:77]: `$out/bin/drawio` will then be unable to find files in the macOS bundle by relative paths.

```
[1021/144723.293589:FATAL:electron_main_delegate_mac.mm(71)] Unable to find helper app
```

Comment (alyssais) [reply pkgs/applications/graphics/drawio/default.nix:77]: Wow, that's horrible.  It could maybe use a comment.

**Review by alyssais — APPROVED** [review]

File: pkgs/applications/graphics/drawio/default.nix [hunk 4]
```diff
@@ -29,13 +28,14 @@ stdenv.mkDerivation rec {
     hash = "sha256-d8AquOKdrPQHBhRG9o1GB18LpwlwQK6ZaM1gLAcjilM=";
   };
 
-  nativeBuildInputs = [
+  nativeBuildInputs = lib.optionals (!stdenv.isDarwin) [
```
Comment (alyssais) [inline pkgs/applications/graphics/drawio/default.nix:31]: I usually see optional dependencies at the end of the list.

File: pkgs/applications/graphics/drawio/default.nix (hunk 3 above)
Comment (wegank) [reply pkgs/applications/graphics/drawio/default.nix:79]: Copied a comment from `utm`.

File: pkgs/applications/graphics/drawio/default.nix (hunk 4 above)
Comment (wegank) [reply pkgs/applications/graphics/drawio/default.nix]: Done. There will then be rebuilds on Linux.

Comment (alyssais) [reply pkgs/applications/graphics/drawio/default.nix]: Better code forever is always preferable to avoiding rebuilds once, IMO.

**Review by alyssais — APPROVED** [review]

**Comment by wegank** [comment]
@ofborg build drawio drawio.passthru.tests

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 262081` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>pandoc-drawio-filter</li>
    <li>pandoc-drawio-filter.dist</li>
    <li>python310Packages.mkdocs-drawio-exporter</li>
    <li>python310Packages.mkdocs-drawio-exporter.dist</li>
    <li>python311Packages.mkdocs-drawio-exporter</li>
    <li>python311Packages.mkdocs-drawio-exporter.dist</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>drawio</li>
  </ul>
</details>
